tests/Extensions/TimedScopes.UnitTests/TimedScopeDefinitionsTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Omex.Extensions.TimedScopes.UnitTests
+{
+	[TestClass]
+	public class TimedScopeDefinitionsTests
+	{
+		[DataTestMethod]
+		[DataRow(null)]
+		[DataRow("")]
+		public void Constructor_ThrowsException(string activityName)
+		{
+			Assert.ThrowsException<ArgumentException>(() =>
+			{
+				new TimedScopeDefinition(activityName);
+			});
+		}
+
+
+		[DataTestMethod]
+		[DataRow("TestName")]
+		public void Constructor_PropagatesName(string activityName)
+		{
+			TimedScopeDefinition definition = new TimedScopeDefinition(activityName);
+			Assert.ReferenceEquals(activityName, definition.Name);
+		}
+
+
+		[TestMethod]
+		public void Equals_WorksProperly()
+		{
+			string name = nameof(Equals_WorksProperly);
+
+			TimedScopeDefinition definition1 = new TimedScopeDefinition(name);
+
+			TimedScopeDefinition definition2 = new TimedScopeDefinition(name);
+
+			Assert.IsTrue(definition1.Equals(definition2), "1 Equals 2");
+
+			Assert.IsTrue(definition2.Equals(definition1), "2 Equals 1");
+
+			Assert.IsTrue(definition1.Equals((object)definition2), "1 Equals (object)2");
+
+			Assert.IsTrue(definition2.Equals((object)definition1), "2 Equals (object)1");
+
+			Assert.IsTrue(definition1.GetHashCode() == definition2.GetHashCode(), "1 HashCode == 2 HashCode");
```

Comment:
For this comparison, it's better to use `Assert.Equal` as it will log the expected and actual in the failure case, making errors easier to diagnose.

Reply:
This asserts is specifically to check `==` and `!=` operators

Reply:
For `GetHashCode()` though, that doesn't make sense. We shouldn't be checking the equality operators for this method. For the other methods though, I agree that what you're doing is right.

Reply:
Oh sorry you are right, I was thinking about a different check, I'll fix it in the follow-up thank you